pgjdbc/src/test/java/org/postgresql/test/jdbc2/BatchedInsertReWriteEnabledTest.java

```diff
@@ -82,47 +126,168 @@ public void testBatchWithReWrittenRepeatedInsertStatementOptimizationEnabled()
+        // "INSERT /*xxx*/INTO testbatch VALUES (?, '1, (, $1234, a''n?d )' /*xxxx)*/, ?) -- xxx",
+    };
+    for (int i = 0; i < odd.length; i++) {
+      String s = odd[i];
+      PreparedStatement pstmt = null;
+      try {
+        pstmt = con.prepareStatement(s);
+        pstmt.setInt(1, 1);
+        pstmt.setInt(2, 2);
+        pstmt.addBatch();
+        pstmt.setInt(1, 3);
+        pstmt.setInt(2, 4);
+        pstmt.addBatch();
+        pstmt.setInt(1, 5);
+        pstmt.setInt(2, 6);
+        pstmt.addBatch();
+        Assert.assertEquals(
+            "Three batched statements should be rewritten to single multi-values one, thus expecting SUCCESS_NO_INFO",
+            Arrays.toString(new int[]{Statement.SUCCESS_NO_INFO,
+                Statement.SUCCESS_NO_INFO, Statement.SUCCESS_NO_INFO}),
+            Arrays.toString(pstmt.executeBatch()));
+      } finally {
+        TestUtil.closeQuietly(pstmt);
       }
     }
   }
 
   /**
    * Test to make sure a statement with a semicolon is not broken
    */
+  private void simpleRewriteBatch(String values, String suffix)
+      throws SQLException {
+    PreparedStatement pstmt = null;
+    try {
+      pstmt = con.prepareStatement("INSERT INTO testbatch " +  values + "(?,?,?)" + suffix);
+      pstmt.setInt(1, 1);
+      pstmt.setString(2, "a");
+      pstmt.setInt(3, 2);
+      pstmt.addBatch();
+      pstmt.setInt(1, 3);
+      pstmt.setString(2, "b");
+      pstmt.setInt(3, 4);
+      pstmt.addBatch();
+      pstmt.setInt(1, 5);
+      pstmt.setString(2, "c");
+      pstmt.setInt(3, 6);
+      pstmt.addBatch();
+      Assert.assertEquals(
+          "Three batched statements should be rewritten to single multi-values one, thus expecting SUCCESS_NO_INFO",
+          Arrays.toString(new int[] { Statement.SUCCESS_NO_INFO,
+              Statement.SUCCESS_NO_INFO, Statement.SUCCESS_NO_INFO }),
+          Arrays.toString(pstmt.executeBatch()));
+    } finally {
+      TestUtil.closeQuietly(pstmt);
+    }
+  }
+
+  /**
+   * Test to make sure a statement with a semicolon is not broken
+   */
+  @Test
   public void testBatchWithReWrittenBatchStatementWithSemiColon()
       throws SQLException {
+    simpleRewriteBatch("values", ";");
+  }
+
+  /**
+   * Test to make sure a statement with a semicolon is not broken
+   */
+  @Test
+  public void testBatchWithReWrittenSpaceAfterValues()
+      throws SQLException {
+    simpleRewriteBatch("values ", "");
+    simpleRewriteBatch("values  ", "");
+    simpleRewriteBatch("values\t", "");
+  }
+
+  /**
+   * Test to make sure a statement with a semicolon is not broken
+   */
+  @Test
+  public void testBatchWithReWrittenMixedCaseValues()
+      throws SQLException {
+    simpleRewriteBatch("vAlues", "");
+    simpleRewriteBatch("vaLUES", "");
+    simpleRewriteBatch("VALUES", "");
+  }
+
+  /**
+   * Test to make sure a statement with a semicolon is not broken
```

Comment:
This is misleading. Test name is fine, yet the comment was copy-pasted from somewhere else.